lib/puppet/pops/parser/epp_support.rb

```diff
@@ -202,6 +202,7 @@ def scan(skip_leading=false)
             @mode = :error
             @issue = Issues::EPP_UNBALANCED_EXPRESSION
           else
+            # rubocop:disable Lint/UselessAssignment
             mode = :epp
```

Comment:
@hlindberg @thallgren this looks incorrect. I assume it should be `@mode = :epp` like is done a few lines above, but this triggers a number of rspec failures. Can you investigate?

Reply:
Aside, this fails CI because I haven't yet enabled the check in .rubocop.yml

Reply:
The logic is for the "end-of-input" case, and it should probably not switch back to `:epp` mode here  and instead stay in the mode it is in - i.e. hitting eof either in `:text` or `:epp` mode. (Confirmed by failing rspec tests).